src/core/rendering.js

```diff
@@ -250,6 +250,85 @@ p5.prototype.createGraphics = function(w, h, renderer) {
   return new p5.Graphics(w, h, renderer, this);
 };
 
+/**
+ * Creates and returns a new <a href="#/p5.Framebuffer">p5.Framebuffer</a>, a
+ * high-performance WebGL object that you can draw to and then use as a texture.
+ *
+ * Options can include:
+ * - `format`: The data format of the texture, either `UNSIGNED_BYTE`, `FLOAT`, or `HALF_FLOAT`. The default is `UNSIGNED_BYTE`.
+ * - `channels`: What color channels to store, either `RGB` or `RGBA`. The default is to match the channels in the main canvas (with alpha unless disabled with `setAttributes`.)
+ * - `depth`: A boolean, whether or not to include a depth buffer. Defaults to true.
+ * - `depthFormat`: The data format for depth information, either `UNSIGNED_BYTE` or `FLOAT`. The default is `UNSIGNED_BYTE` if `format` is that as well, or `FLOAT` otherwise.
```

Comment:
Sorry I can't remember where we landed on this, but did we discover that unsigned_byte is only used for the depth buffer in cases where it's a stencil? Would it be better to default to float for depth buffers?

Reply:
I think the first three rows of this table are the options that we have available to us:

![image](https://user-images.githubusercontent.com/5315059/228544053-b17ec7c1-b8cc-4888-8e0c-0f3e7566bd7c.png)

The first two are unsigned integers, either 16 or 24 bits. I guess if we want to be accurate we should call the deoth format `UNSIGNED_INT` (since that's the format that we're using internally anyway)? https://github.com/processing/p5.js/blob/4cfcab87ebcd89a9cb2d5d2b13e9ba4ea02a9eba/src/webgl/p5.Framebuffer.js#L601-L605

For defaulting to float, that could make sense, since it's prooobably not worse than integers and is nicer when reading values? The reason why I it's an int right now by default is that that's the default for a WebGL canvas (and p5.Graphics) with no framebuffer: https://registry.khronos.org/webgl/specs/latest/1.0/#2.2

so, I guess questions for us to answer:
- Should we make an `UNSIGNED_INT` option just for depth? (Additionally, we could maybe use `UNSIGNED_INT` for color too, since `UNSIGNED_BYTE` is an `UNSIGNED_INT`, just with a size specified, and then we'd have closer to the same options for color and depth.)
- Do you think it's better to match the depth format of p5.Graphics and use an int or use the probably more useful format and use float?

Reply:
> Should we make an UNSIGNED_INT option just for depth? (Additionally, we could maybe use UNSIGNED_INT for color too, since UNSIGNED_BYTE is an UNSIGNED_INT, just with a size specified, and then we'd have closer to the same options for color and depth.)

Making a UNSIGNED_INT option for depth sounds right to me. And also is maybe more descriptive of a value since...they are ints.

> Do you think it's better to match the depth format of p5.Graphics and use an int or use the probably more useful format and use float?

My gut feeling says it'd be better to use the float format _if_ it is available to us.